src/nvim/CMakeLists.txt

```diff
@@ -120,9 +120,13 @@ if(NOT MSVC)
   set_source_files_properties(
     ${CONV_SOURCES} PROPERTIES COMPILE_FLAGS "${COMPILE_FLAGS} -Wno-conversion")
   # gperf generates ANSI-C with incorrect linkage, ignore it.
+  check_c_compiler_flag(-Wno-static-in-inline HAS_WNO_STATIC_IN_INLINE_FLAG)
+  if(HAS_WNO_STATIC_IN_INLINE_FLAG)
+    set_source_files_properties(
+      eval.c PROPERTIES COMPILE_FLAGS "${COMPILE_FLAGS} -Wno-static-in-inline")
+  endif()
   set_source_files_properties(
-    eval.c PROPERTIES COMPILE_FLAGS "${COMPILE_FLAGS} -Wno-static-in-inline -Wno-conversion")
-
+    eval.c PROPERTIES COMPILE_FLAGS "${COMPILE_FLAGS} -Wno-conversion")
```

Comment:
I expected the second **set** to include the current **C_COMPILE_FLAGS** + the new option. Lets see, in OpenBSD i get


[ 58%] Building C object src/nvim/CMakeFiles/nvim.dir/eval.c.o
cd /home/raf/Code/neovim/build/src/nvim && /usr/bin/cc  -DHAVE_CONFIG_H -DINCLUDE_GENERATED_DECLARATIONS -I/home/raf/Code/neovim/build/config -I/home/raf/Code/neovim/src -isystem /home/raf/Code/neovim/.deps/usr/include -isystem /usr/local/include -I/home/raf/Code/neovim/build/src/nvim/auto -I/home/raf/Code/neovim/build/include  -Wconversion -O2 -g   -Wall -Wextra -pedantic -Wno-unused-parameter -Wstrict-prototypes -std=gnu99 -fstack-protector-strong  **-Wno-conversion** -o CMakeFiles/nvim.dir/eval.c.o   -c /home/raf/Code/neovim/src/nvim/eval.c

that seems correct. In Linux I get

[ 50%] Building C object src/nvim/CMakeFiles/nvim.dir/eval.c.o
cd /home/raf/Code/neovim/build/src/nvim && /usr/bin/cc  -DHAVE_CONFIG_H -DINCLUDE_GENERATED_DECLARATIONS -D_GNU_SOURCE -I/home/raf/Code/neovim/build/config -I/home/raf/Code/neovim/src -isystem /home/raf/Code/neovim/.deps/usr/include -I/home/raf/Code/neovim/build/src/nvim/auto -I/home/raf/Code/neovim/build/include  -Wconversion -Og -g   -Wall -Wextra -pedantic -Wno-unused-parameter -Wstrict-prototypes -std=gnu99 -Wvla -fstack-protector-strong -fdiagnostics-color=auto  **-Wno-conversion** -o CMakeFiles/nvim.dir/eval.c.o   -c /home/raf/Code/neovim/src/nvim/eval.c

which is missing Wno-static-in-inline. Pushing a fix shortly.